--- xen-4.3.1/Config.mk.orig	2013-10-30 14:22:56.000000000 +0000
+++ xen-4.3.1/Config.mk	2014-12-02 22:55:33.907514300 +0000
@@ -219,7 +219,7 @@
 # Wed Jun 26 16:30:45 2013 +0100
 # xen: Don't perform SMP setup.
 
-ETHERBOOT_NICS ?= rtl8139 8086100e
+ETHERBOOT_NICS ?= 10ec8139 8086100e
 
 # Specify which qemu-dm to use. This may be `ioemu' to use the old
 # Mercurial in-tree version, or a local directory, or a git URL.
--- xen-4.3.1/tools/firmware/Makefile.orig	2013-10-30 08:18:28.000000000 +0000
+++ xen-4.3.1/tools/firmware/Makefile	2014-12-02 22:57:06.191313600 +0000
@@ -10,7 +10,7 @@
 SUBDIRS-$(CONFIG_SEABIOS) += seabios-dir
 SUBDIRS-$(CONFIG_ROMBIOS) += rombios
 SUBDIRS-$(CONFIG_ROMBIOS) += vgabios
-SUBDIRS-$(CONFIG_ROMBIOS) += etherboot
+#SUBDIRS-$(CONFIG_ROMBIOS) += etherboot
 SUBDIRS-y += hvmloader
 
 ovmf:
--- xen-4.3.1/tools/firmware/hvmloader/Makefile.orig	2013-10-30 08:18:28.000000000 +0000
+++ xen-4.3.1/tools/firmware/hvmloader/Makefile	2014-12-02 22:54:15.065301400 +0000
@@ -48,7 +48,8 @@
 else
 CIRRUSVGA_ROM := ../vgabios/VGABIOS-lgpl-latest.cirrus.bin
 endif
-ETHERBOOT_ROMS := $(addprefix ../etherboot/ipxe/src/bin/, $(addsuffix .rom, $(ETHERBOOT_NICS)))
+ETHERBOOT_ROMS := $(addprefix /usr/share/ipxe/, $(addsuffix .rom, $(ETHERBOOT_NICS)))
+
 endif
 
 ROMS := 
